src/main.cpp

```diff
@@ -1520,6 +1520,10 @@ bool AcceptToMemoryPoolWorker(CTxMemPool& pool, CValidationState& state, const C
                 __func__, hash.ToString(), FormatStateMessage(state));
         }
 
+        // Check for non-standard witness in P2WSH after transaction is validated
```

Comment:
It's about the #8279. But since this is just one of the many ways to mutate the witness and we won't ban the peer anyway, maybe it makes more sense to move it before validation and after `AreInputsStandard`?